packages/front-end/components/Settings/EditDataSource/DataSourceExperimentProperties/DataSourceViewEditExperimentProperties.tsx

```diff
@@ -14,9 +16,14 @@ export const DataSourceViewEditExperimentProperties: FC<DataSourceViewEditExperi
   onSave,
   onCancel,
   dataSource,
+  canEdit = true,
 }) => {
+  const { project } = useDefinitions();
   const [uiMode, setUiMode] = useState<"view" | "edit">("view");
 
+  const permissions = usePermissions();
+  canEdit = canEdit && permissions.check("editDatasourceSettings", project);
```

Comment:
When editing an existing data source, it should use the datasource's projects the check permissions, not the one from definitions.